File: codesignal/arcade/2.core/7.book market/6.HTMLEndTagByStartTag.cpp
// You are implementing your own HTML editor. To make it more comfortable for developers you 
// would like to add an auto-completion feature to it.

// Given the starting HTML tag, find the appropriate end tag which your editor should propose.

#include <string>

class Solution {
public:
  std::string htmlEndTagByStartTag(std::string startTag) {
    int len = startTag.length();
    int i = 1;
    while (i < len && startTag[i] != ' ' && startTag[i] != '>') i++;
    return "</" + startTag.substr(1, i - 1) + ">";
  }
};

File: codesignal/arcade/2.core/7.book market/6.HTMLEndTagByStartTag.py
# You are implementing your own HTML editor. To make it more comfortable for developers you 
# would like to add an auto-completion feature to it.

# Given the starting HTML tag, find the appropriate end tag which your editor should propose

class Solution:
    def HTMLEndTagByStartTag(self, startTag):
        return "</" + startTag[1:-1].split()[0] + ">"
    